src/CredentialSource/AwsNativeSource.php

```diff
@@ -0,0 +1,314 @@
+<?php
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth\CredentialSource;
+
+use Google\Auth\ExternalAccountCredentialSourceInterface;
+use Google\Auth\HttpHandler\HttpClientCache;
+use Google\Auth\HttpHandler\HttpHandlerFactory;
+use GuzzleHttp\Psr7\Request;
+
+/**
+ * Authenticates requests using IAM credentials.
+ */
+class AwsNativeSource implements ExternalAccountCredentialSourceInterface
+{
+    private const CRED_VERIFICATION_QUERY = 'Action=GetCallerIdentity&Version=2011-06-15';
+
+    private string $audience;
+    private string $regionUrl;
+    private string $regionalCredVerificationUrl;
+    private ?string $securityCredentialsUrl;
+
+    public function __construct(
+        string $audience,
+        string $regionUrl,
+        string $regionalCredVerificationUrl,
+        string $securityCredentialsUrl = null
+    ) {
+        $this->audience = $audience;
+        $this->regionUrl = $regionUrl;
+        $this->regionalCredVerificationUrl = $regionalCredVerificationUrl;
+        $this->securityCredentialsUrl = $securityCredentialsUrl;
+    }
+
+    public function fetchSubjectToken(callable $httpHandler = null): string
+    {
+        if (is_null($httpHandler)) {
+            $httpHandler = HttpHandlerFactory::build(HttpClientCache::getHttpClient());
+        }
+
+        $awsToken = self::fetchAwsTokenFromMetadata($httpHandler);
+
+        $signingVars = $this->securityCredentialsUrl
```

Comment:
Nit: to be consistent with the other libraries, we should be getting the security credentials from the environment variables, even if the url has been provided in the credential configuration.

Reply:
Got it, so rather than "If securityCredentialsUrl then use the URL, otherwise use env vars", the logic should be "If env vars, use them, otherwise use the security credentials URL"?

Reply:
(this should now be fixed)